toolchain/check/check.cpp

```diff
@@ -253,40 +254,49 @@ static auto ImportOtherPackages(Context& context, UnitInfo& unit_info,
 
   for (auto [i, api_imports_entry] : llvm::enumerate(api_imports_list)) {
     // These variables are updated after figuring out which imports are present.
-    Parse::ImportDeclId node_id = Parse::InvalidNodeId();
+    auto import_decl_id = SemIR::InstId::Invalid;
     IdentifierId package_id = IdentifierId::Invalid;
     bool has_load_error = false;
 
     // Identify the local package imports if present.
-    const UnitInfo::PackageImports* local_imports = nullptr;
+    UnitInfo::PackageImports* local_imports = nullptr;
     if (i < unit_info.package_imports.size()) {
       local_imports = &unit_info.package_imports[i];
       if (!local_imports->package_id.is_valid()) {
         // Skip the current package.
         continue;
       }
-      node_id = local_imports->node_id;
+      import_decl_id = local_imports->import_decl_id;
+
       package_id = local_imports->package_id;
       has_load_error |= local_imports->has_load_error;
     }
 
     // Identify the API package imports if present.
-    const UnitInfo::PackageImports* api_imports = nullptr;
+    UnitInfo::PackageImports* api_imports = nullptr;
     if (api_imports_entry.second != -1) {
       api_imports =
           &unit_info.api_for_impl->package_imports[api_imports_entry.second];
-      if (!package_id.is_valid()) {
-        package_id = api_imports_entry.first;
-      } else {
+
+      if (local_imports) {
         CARBON_CHECK(package_id == api_imports_entry.first);
+      } else {
+        auto import_ir_inst_id = context.import_ir_insts().Add(
+            {.ir_id = SemIR::ImportIRId::ApiForImpl,
+             .inst_id = api_imports->import_decl_id});
+        import_decl_id = context.AddInst<SemIR::ImportDecl>(
+            import_ir_inst_id, {.package_id = SemIR::NameId::ForIdentifier(
+                                    api_imports_entry.first)});
```

Comment:
I did, but right now we're only using `ImportRef` for entities. That is, `ImportRef` points at a typed entity in name lookup.

While imports from other packages can create a namespace that's in name lookup, the import declaration itself isn't in name lookup, and the current package's imports don't have any name lookup data. So it felt like this approach, closer to `Namespace` handling than entity handling, was a better fit.

Also, presumably it'd be an `ImportRefLoaded` (`ImportRefUnloaded` feels a little odd to me since we do load the import), and that `ImportRefLoaded` should be linked to a resolved `ImportDecl` and have type information. Building that would require defining a type for `ImportDecl`, and I'm not sure what that would be (maybe engaging in a more circular definition of `ImportDecl` as a back-reference to the `Namespace`?)